python/plugins/processing/algs/gdal/DatasetIdentify.py

```diff
@@ -0,0 +1,124 @@
+*   the Free Software Foundation; either version 2 of the License, or     *
+*   (at your option) any later version.                                   *
+*                                                                         *
+***************************************************************************
+"""
+
+__author__ = "Even Rouault"
+__date__ = "December 2025"
+__copyright__ = "(C) 2025, Even Rouault"
+
+import os
+
+from qgis.core import (
+    Qgis,
+    QgsProcessing,
+    QgsProcessingException,
+    QgsProcessingParameterFile,
+    QgsProcessingParameterBoolean,
+    QgsProcessingParameterVectorDestination,
+)
+from processing.algs.gdal.GdalAlgorithm import GdalAlgorithm
+from processing.algs.gdal.GdalUtils import GdalUtils
+
+pluginPath = os.path.split(os.path.split(os.path.dirname(__file__))[0])[0]
+
+
+class DatasetIdentify(GdalAlgorithm):
+    INPUT = "INPUT"
+    RECURSIVE = "RECURSIVE"
+    DETAILS = "DETAILS"
+    OUTPUT = "OUTPUT"
+
+    def __init__(self):
+        super().__init__()
+
+    def initAlgorithm(self, config=None):
+        self.addParameter(
+            QgsProcessingParameterFile(
+                self.INPUT,
+                self.tr("Input folder"),
+                Qgis.ProcessingFileParameterBehavior.Folder,
+            )
+        )
+
+        self.addParameter(
+            QgsProcessingParameterBoolean(
+                self.RECURSIVE,
+                self.tr(
+                    "Perform recursive exploration of the input folder and its subfolders"
+                ),
+                defaultValue=True,
+            )
+        )
+
+        self.addParameter(
+            QgsProcessingParameterBoolean(
+                self.DETAILS,
+                self.tr("Add details about identified datasets in the output"),
+                defaultValue=True,
+            )
+        )
+
+        self.addParameter(
+            QgsProcessingParameterVectorDestination(
+                self.OUTPUT,
+                self.tr("Output file"),
+                QgsProcessing.SourceType.TypeVector,
+            )
+        )
+
+    def name(self):
+        return "DatasetIdentify"
+
+    def displayName(self):
+        return self.tr("Dataset identification")
+
+    def group(self):
+        return self.tr("Miscellaneous")
+
+    def groupId(self):
+        return "miscellaneous"
+
+    def commandName(self):
+        return "gdal dataset identify"
+
+    def getConsoleCommands(self, parameters, context, feedback, executing=True):
+        arguments = []
+        if self.parameterAsBoolean(parameters, self.RECURSIVE, context):
+            arguments.append("--recursive")
+        if self.parameterAsBoolean(parameters, self.DETAILS, context):
+            arguments.append("--detailed")
+
+        inFile = self.parameterAsFile(parameters, self.INPUT, context)
+        if inFile is None:
+            raise QgsProcessingException(
+                self.invalidRasterError(parameters, self.INPUT)
+            )
```

Comment:
This seems a bit misleading to me, input is a folder but we report it as an invalid raster. Maybe it would be better to do something like

```suggestion
        if not os.path.exists(inFile):
            raise QgsProcessingException(
                self.tr("Directory {dirName} does not exists!".format(dirName=inFile))
            )
```